Index: slmodem-2.9.11-20080817/modem/modem.h
===================================================================
--- slmodem-2.9.11-20080817.orig/modem/modem.h	2008-03-29 14:04:16.000000000 +0200
+++ slmodem-2.9.11-20080817/modem/modem.h	2008-09-17 15:17:49.000000000 +0200
@@ -301,7 +301,7 @@
 	void *cid;
 #endif
         /* dialer */
-        char dial_string[32];
+        char dial_string[128];
         /* escape counter */
         unsigned escape_count;
 	unsigned long last_esc_check;
