
Index: slmodem-2.9.11-20080817/modem/modem.c
===================================================================
--- slmodem-2.9.11-20080817.orig/modem/modem.c	2008-03-29 14:04:16.000000000 +0200
+++ slmodem-2.9.11-20080817/modem/modem.c	2008-09-17 15:18:09.000000000 +0200
@@ -1696,6 +1696,7 @@
 	return 0;
 }
 
+static int sregs_init(unsigned char sregs[]);
 
 int modem_reset(struct modem *m)
 {
