Index: sl-modem-2.9.11~20080817/drivers/st7554.c
===================================================================
--- sl-modem-2.9.11~20080817.orig/drivers/st7554.c	2009-03-20 15:03:49.000000000 +0200
+++ sl-modem-2.9.11~20080817/drivers/st7554.c	2009-03-20 15:03:49.000000000 +0200
@@ -58,13 +58,6 @@
 
 #include <modem_defs.h>
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,13)
-#define class_simple_device_add(class, dev, addr, name, i)
-#define class_simple_device_remove(dev)
-#define class_simple_create(module, name) (struct class_simple *)(name)
-#define class_simple_destroy(class)
-#endif
-
 #define MAX_MODEMS 16
 
 #define USB_INFO(fmt...) printk(KERN_INFO fmt)
Index: sl-modem-2.9.11~20080817/drivers/amrmo_init.c
===================================================================
--- sl-modem-2.9.11~20080817.orig/drivers/amrmo_init.c	2009-03-20 15:03:49.000000000 +0200
+++ sl-modem-2.9.11~20080817/drivers/amrmo_init.c	2009-03-20 15:09:46.000000000 +0200
@@ -75,10 +75,11 @@
 #endif
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,13)
-#define class_simple_device_add(class, dev, addr, name, i)
-#define class_simple_device_remove(dev)
-#define class_simple_create(module, name) (struct class_simple *)(name)
-#define class_simple_destroy(class)
+#define class_simple_device_add(class, dev, addr, name, i) device_create(class, dev, addr, name, i)
+#define class_simple_device_remove(class, dev) device_destroy(class, dev)
+#define class_simple_create(module, name) class_create(module, name)
+#define class_simple_destroy(class) class_destroy(class)
+static struct class *amrmo_class;
 #else
 #define pci_match_id(tbl, dev) pci_match_device(tbl, dev)
 #endif
@@ -318,7 +319,7 @@
 
 static struct amrmo_struct *amrmo_table[MAXNUM] = {};
 #ifndef OLD_KERNEL
-static struct class_simple *amrmo_class;
+//static struct class_simple *amrmo_class;
 #endif
 
 /*
@@ -699,7 +700,7 @@
 	}
 #endif
 #else
-	class_simple_device_remove(MKDEV(AMRMO_MAJOR, amrmo->num));
+	class_simple_device_remove(amrmo_class, MKDEV(AMRMO_MAJOR, amrmo->num));
 #endif
 	amrmo_table[amrmo->num] = NULL;
 	amrmo_card_disable(amrmo->card);
@@ -732,7 +733,7 @@
 
 MODULE_AUTHOR("Smart Link Ltd.");
 MODULE_DESCRIPTION("SmartLink HAMR5600,SmartPCI56/561 based modem driver");
-MODULE_LICENSE("Smart Link Ltd.");
+MODULE_LICENSE("Dual BSD/GPL");
 MODULE_ALIAS_CHARDEV_MAJOR(AMRMO_MAJOR);
 
 
