Index: debian/changelog
===================================================================
--- debian/changelog	(Revision 1905)
+++ debian/changelog	(Arbeitskopie)
@@ -1,3 +1,11 @@
+sl-modem (2.9.9d-7) UNRELEASED; urgency=low
+
+  * NOT RELEASED YET
+  * http://dev.gentoo.org/~genstef/files/slmodem-class-simple-to-class.diff
+    added (closes: ...)
+
+ -- Eduard Bloch <blade@debian.org>  Mon, 15 Aug 2005 17:17:37 +0200
+
 sl-modem (2.9.9d-6) unstable; urgency=low
 
   * applied gcc-4.0 fixes from Sasha Khapyorsky, dropped gcc-3.4 workarounds
Index: drivers/amrmo_init.c
===================================================================
--- drivers/amrmo_init.c	(Revision 1810)
+++ drivers/amrmo_init.c	(Arbeitskopie)
@@ -283,8 +283,20 @@
 
 static struct amrmo_struct *amrmo_table[MAXNUM] = {};
 #ifndef OLD_KERNEL
+#ifdef FOUND_CLASS_SIMPLE
+#define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) class_simple_device_add(class, dev, device, fmt, rest)
+#define CLASS_DESTROY(class) class_simple_destroy(class)
+#define CLASS_DEVICE_DESTROY(class, dev) class_simple_device_remove(dev)
+#define CLASS_CREATE(owner, name) class_simple_create(owner, name)
 static struct class_simple *amrmo_class;
+#else
+#define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) class_device_create(class, dev, device, fmt, rest)
+#define CLASS_DESTROY(class) class_destroy(class)
+#define CLASS_DEVICE_DESTROY(class, dev) class_device_destroy(class, dev)
+#define CLASS_CREATE(owner, name) class_create(owner, name)
+static struct class *amrmo_class;
 #endif
+#endif
 
 /*
  *    debug stuff
@@ -629,7 +641,7 @@
 	}
 #endif
 #else
-	class_simple_device_add(amrmo_class, MKDEV(AMRMO_MAJOR, i), NULL, "slamr%d", i);
+	CLASS_DEVICE_CREATE(amrmo_class, MKDEV(AMRMO_MAJOR, i), NULL, "slamr%d", i);
 	devfs_mk_cdev(MKDEV(AMRMO_MAJOR,i), S_IFCHR|S_IRUSR|S_IWUSR, "slamr%d", i);
 #endif
 	return 0;
@@ -661,7 +673,7 @@
 	}
 #endif
 #else
-	class_simple_device_remove(MKDEV(AMRMO_MAJOR, amrmo->num));
+	CLASS_DEVICE_DESTROY(amrmo_class, MKDEV(AMRMO_MAJOR, amrmo->num));
 	devfs_remove("slamr%d", amrmo->num);
 #endif
 	amrmo_table[amrmo->num] = NULL;
@@ -745,7 +757,7 @@
 	}
 
 #ifndef OLD_KERNEL
-	amrmo_class = class_simple_create(THIS_MODULE, "slamr");
+	amrmo_class = CLASS_CREATE(THIS_MODULE, "slamr");
 	if (IS_ERR(amrmo_class)) {
 		int err = PTR_ERR(amrmo_class);
 		printk(KERN_ERR "slamr: failure creating simple class, error %d\n", err);
@@ -755,7 +767,7 @@
 	if ((err = pci_register_driver(&amrmo_pci_driver)) < 0) {
 		pci_unregister_driver(&amrmo_pci_driver);
 #ifndef OLD_KERNEL
-		class_simple_destroy(amrmo_class);
+		CLASS_DESTROY(amrmo_class);
 #endif
                 return err;
 	}
@@ -763,7 +775,7 @@
 	if(register_chrdev(AMRMO_MAJOR, "slamr", &amrmo_fops) < 0) {
 		pci_unregister_driver(&amrmo_pci_driver);
 #ifndef OLD_KERNEL
-		class_simple_destroy(amrmo_class);
+		CLASS_DESTROY(amrmo_class);
 #endif
 		return -ENOMEM;
 	}
@@ -777,7 +789,7 @@
 	unregister_chrdev(AMRMO_MAJOR,"slamr");
 	pci_unregister_driver(&amrmo_pci_driver);
 #ifndef OLD_KERNEL
-	class_simple_destroy(amrmo_class);
+	CLASS_DESTROY(amrmo_class);
 #endif
 }
 
Index: drivers/st7554.c
===================================================================
--- drivers/st7554.c	(Revision 1810)
+++ drivers/st7554.c	(Arbeitskopie)
@@ -206,7 +206,19 @@
 
 
 static struct st7554_state *st7554_table[MAX_MODEMS] = {};
+#ifdef FOUND_CLASS_SIMPLE
+#define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) class_simple_device_add(class, dev, device, fmt, rest)
+#define CLASS_DESTROY(class) class_simple_destroy(class)
+#define CLASS_DEVICE_DESTROY(class, dev) class_simple_device_remove(dev)
+#define CLASS_CREATE(owner, name) class_simple_create(owner, name)
 static struct class_simple *st7554_class;
+#else
+#define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) class_device_create(class, dev, device, fmt, rest)
+#define CLASS_DESTROY(class) class_destroy(class)
+#define CLASS_DEVICE_DESTROY(class, dev) class_device_destroy(class, dev)
+#define CLASS_CREATE(owner, name) class_create(owner, name)
+static struct class *st7554_class;
+#endif
 
 static DECLARE_MUTEX(open_sem);
 
@@ -1271,7 +1283,7 @@
 	}
 
 	usb_set_intfdata(interface, s );
-	class_simple_device_add(st7554_class, MKDEV(243, i), NULL, "slusb%d", i);
+	CLASS_DEVICE_CREATE(st7554_class, MKDEV(243, i), NULL, "slusb%d", i);
 	devfs_mk_cdev(MKDEV(243,i),S_IFCHR|S_IRUSR|S_IWUSR,"slusb%d",i);
 
 	USB_INFO(KERN_INFO "slusb: slusb%d is found.\n", s->minor);
@@ -1300,7 +1312,7 @@
                 return;
         }
 
-	class_simple_device_remove(MKDEV(243, s->minor));
+	CLASS_DEVICE_DESTROY(st7554_class, MKDEV(243, s->minor));
  	devfs_remove("slusb%d",s->minor);
 
 	st7554_stop(s);
@@ -1334,7 +1346,7 @@
 	int ret;
 	USB_INFO ("ST7554 USB Modem.\n");
 
-	st7554_class = class_simple_create(THIS_MODULE, "slusb");
+	st7554_class = CLASS_CREATE(THIS_MODULE, "slusb");
 	if (IS_ERR(st7554_class)) {
 		ret = PTR_ERR(st7554_class);
 		USB_ERR("st7554_modem_init: failed to create sysfs class, error %d\n", ret);
@@ -1344,13 +1356,13 @@
 	ret = usb_register(&st7554_usb_driver);
 	if ( ret ) {
 		USB_ERR ("st7554_modem_init: cannot register usb device.\n");
-		class_simple_destroy(st7554_class);
+		CLASS_DESTROY(st7554_class);
 		return ret;
 	}
 
 	if(register_chrdev(243, "slusb", &st7554_fops) < 0) {
 		usb_deregister(&st7554_usb_driver);
-		class_simple_destroy(st7554_class);
+		CLASS_DESTROY(st7554_class);
 		return -ENOMEM;
 	}
 	return 0;
@@ -1362,7 +1374,7 @@
 	USB_DBG ("st7554: exit...\n");
 	unregister_chrdev(243,"slusb");
 	usb_deregister(&st7554_usb_driver);
-	class_simple_destroy(st7554_class);
+	CLASS_DESTROY(st7554_class);
 }
 
 
Index: drivers/Makefile
===================================================================
--- drivers/Makefile	(Revision 1883)
+++ drivers/Makefile	(Arbeitskopie)
@@ -18,8 +18,10 @@
 
 KERNEL_DIR:=/lib/modules/$(shell uname -r)/build
 
-EXTRA_CFLAGS = -I$(obj) -I$(obj)/../modem
+FOUND_CLASS_SIMPLE := $(shell grep -q 'class_simple_device_add' ${KERNEL_DIR}/include/linux/device.h 2> /dev/null && echo -DFOUND_CLASS_SIMPLE)
 
+EXTRA_CFLAGS = -I$(obj) -I$(obj)/../modem $(FOUND_CLASS_SIMPLE)
+
 obj-m := slamr.o slusb.o
 
 slamr-objs:= amrmo_init.o sysdep_amr.o amrlibs.o
@@ -69,7 +71,7 @@
 module-dir ?= ${DESTDIR}/lib/modules/$(KERNEL_VER)/misc
 
 
-CFLAGS:= -Wall -pipe -O3 -fomit-frame-pointer -D__KERNEL__ -DMODULE -DEXPORT_SYMTAB `test -f $(KERNEL_DIR)/include/linux/modversions.h && echo -DMODVERSIONS --include $(KERNEL_DIR)/include/linux/modversions.h -I$(KERNEL_DIR)/include`
+CFLAGS:= -Wall -pipe -O3 -fomit-frame-pointer -D__KERNEL__ -DMODULE -DEXPORT_SYMTAB `test -f $(KERNEL_DIR)/include/linux/modversions.h && echo -DMODVERSIONS --include $(KERNEL_DIR)/include/linux/modversions.h -I$(KERNEL_DIR)/include` $(FOUND_CLASS_SIMPLE)
 
 all: $(obj-m)
 
